### PR TITLE
Pull Request Nando

### PR DESCRIPTION
**Pull request by nandoooooooo**
Added a new `median` function to calculate the median of a number array, along with improved error handling for empty arrays in the `mean` function. Expanded test coverage for `mean`, `median`, and new calculator operations (`multiply` and `divide`), including edge cases like division by zero.

### CODE REVIEW

File: src/average.test.ts [hunk 1]
```diff
@@ -1,13 +1,67 @@
...
+
+  // Assert
+  expect(actual).toBe(expected);
+});
+
+test("test median for even number of elements", () => {
+  // Arrange
+  const numbers: number[] = [-1231, 2, 4130, 123123];
+  const sortedNumbers = [...numbers].sort((a, b) => a - b);
+  const expected: number = (sortedNumbers[1] + sortedNumbers[2]) / 2;
```
Comment (patrickbucher) [inline src/average.test.ts:51]: Den Testcode könnte man auch "dümmer" machen; hier hat man praktisch eine zweite Implementierung als Test.

**Comment by patrickbucher** [comment]
Gut; siehe Kommentar.
